renderdoc/driver/d3d11/d3d11_initstate.cpp

```diff
@@ -377,20 +386,27 @@ uint32_t WrappedID3D11Device::GetSize_InitialState(ResourceId id, ID3D11DeviceCh
     ret += 4;                      // number of subresources
     ret += 8 * NumSubresources;    // RowPitch and DepthPitch for each subresource
 
+    ID3D11Resource *stage = (ID3D11Resource *)m_ResourceManager->GetInitialContents(id).resource;
+
     // Subresource contents:
     for(UINT sub = 0; sub < NumSubresources; sub++)
     {
       UINT mip = GetMipForSubresource(tex, sub);
 
+      if(stage)
+        pitch = GetResourcePitchForSubresource(m_pImmediateContext->GetReal(), stage, sub);
+      else
+        pitch = {WorstRowPitchAlign, WorstDepthPitchAlign};
+
       uint32_t numRows = RDCMAX(1U, desc.Height >> mip);
       if(IsBlockFormat(desc.Format))
         numRows = AlignUp4(numRows) / 4;
 
       const UINT RowPitch = GetByteSize(desc.Width, 1, 1, desc.Format, mip);
-      const UINT WorstRowPitch = AlignUp(RowPitch, WorstRowPitchAlign);
+      const UINT WorstRowPitch = AlignUp(RowPitch, pitch.m_RowPitch);
 
       const UINT DepthPitch = WorstRowPitch * numRows;
-      const UINT WorstDepthPitch = AlignUp(DepthPitch, WorstDepthPitchAlign);
+      const UINT WorstDepthPitch = AlignUp(DepthPitch, pitch.m_DepthPitch);
```

Comment:
Looking at this again now, I'm not sure I understand the logic here.

If we have the actual row/depth pitch they should just be used directly, not applied as an alignment on the calculated tightly-packed row width. In many cases the row or depth pitch from the driver won't be a power of two so the align calculation won't give the result we want.

The only reason I had 'aligning' here before was because I needed some way to estimate how much padding there would be on top of the tightly packed calculation of how many bytes a row contained at minimum. Since I was mostly concerned about alignment of the rows or padding for small mips, I added a pessimistic alignment to apply.

Reply:
Yep, you're right. That is stupid naive code. Will patch that up.